src/main/java/sam/Sam.java

```diff
@@ -0,0 +1,228 @@
+package sam;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * The Sam class represents a task management application.
+ * It allows users to add, mark, unmark, and delete tasks.
+ * Tasks can be of type ToDo, Deadline, or Event.
+ * The application provides a command-line interface for users to interact with.
+ * Users can input commands to perform various operations on the tasks.
+ * The application also supports saving and loading tasks from a file.
+ *
+ * The Sam class contains methods to handle user commands and perform the corresponding operations.
+ * It uses the Ui class to display messages to the user.
+ * It uses the Storage class to save and load tasks from a file.
+ * It uses the Items class to store and manage the list of tasks.
+ *
+ * The main method creates an instance of the Sam class and starts the application.
+ *
+ */
+public class Sam {
+    private Ui ui;
+    private Storage storage;
+    private Items items;
+
+    /**
+     * Constructs a new instance of the Sam class.
+     *
+     * @param filePath The file path to the storage file.
+     */
+    public Sam() {
+        String filePath = "data/Sam.txt";
```

Comment:
I disagree with @arnav12344 's comment here. Since it is only used in once in the entire program, I think it can just be hardcoded there:
```
storage = new Storage("data/Sam.txt");
```